prov/gni/src/gnix_ep.c

```diff
@@ -276,15 +279,21 @@ static int gnix_ep_close(fid_t fid)
 {
 	struct gnix_ep *ep;
 	struct gnix_domain *domain;
+	struct gnix_nic *nic;
 
 	ep = container_of(fid, struct gnix_ep, ep_fid.fid);
 	/* TODO: lots more stuff to do here */
 
 	domain = ep->domain;
 	assert(domain != NULL);
-
 	atomic_dec(&domain->ref_cnt);
 
+	nic = ep->nic;
+	assert(nic != NULL);
+	atomic_dec(&nic->ref_cnt);
+
+	memset(ep,0,sizeof *ep);
```

Comment:
Is this just being paranoid or is there a another reason to clear the structure before freeing it?